### DISCUSSION TITLE
Mini 11 USB Power Trigger?

### DISCUSSION THREAD

**Discussion by pganderton** [gopro/labs#335] [Q&A]
I've been successfully using a Mini 11 with the USB Power Trigger to start/stop recording with USB power as the trigger, but this is not now working on a new Mini11 with the latest or previous FW. I now see that this function is listed as not available for the mini in the 'overview of labs functionality' table. Is this correct and might there be a workaround? Basically I just need to only record when USB power is present, repeating multiple times over many hours.
Thanks in advance for any help,
Peter

**Comment by dnewman-gpsw** [comment]
The old method is obsolete. There is way more control now https://youtu.be/T87RekjaVA4

Now that the new controls are available for MAX, H11 and Mini, I need to update the power trigger documentation.  

The new methods are more complex, but vasty more configurable and reliable.

Steps:

!MWAKE=2  - this makes the camera boot when power is added.

!MBOOT="!Lboot"  -  this makes the camera run custom commands stored on the SD card within a file called MISC/boot.

Now for the clever new parts, all customized to your needs. This is simple version for Mini
```
!SAVEboot=<u0!X>u0<r0!S<u0>r0!E!R5  - this saves the "boot" script we the new custom triggers.
```
Explained in pseudo code

```
   <u0!X     - if(usb_power < 0) Exit();  // use camera as normal
loop:
   >u0<r0!S  - if(usb_power on AND not record) StartRecording();
   <u0>r0!S  - if(usb_power off AND recording) StopRecording();
   !R5       - goto loop;  // runs until manual power off
```

As single QR Code:
![image](https://user-images.githubusercontent.com/10856084/236506377-2af6a90d-d551-4b42-8f84-f2e4f4ef9df8.png)

To charge the camera, eject the sd card before adding power. As the custom boot commands are on the SD, the camera will behave more stock.

**Reply by spdavidb** [reply]
How would this code be amended to auto power off when USB power is disconnected?

**Reply by dnewman-gpsw** [reply]
It can be yes. 

Same:
!MWAKE=2!MBOOT="!Lboot" 

Try changing:
!SAVEboot=<u0!X>u0<r0!S>r0!E+!1O!R5 

Adds the !1O - shutdown after ending the capture.

**Reply by spdavidb** [reply]
Thank you, getting closer, but getting some unexpected behavior with this one.

Using: !MWAKE=2!MBOOT="!Lboot"!SAVEboot=<u0!X>u0<r0!S>r0!E+!1O!R5

Camera boots on USB power, records for :02, then stops recording, onscreen text "Pause for 1 secs button to cancel" and camera powers off, then when USB power disconnects it powers back up.

**Reply by dnewman-gpsw** [reply]
That is what that script will do. The '>r0' should be '<u0' is you want to record to stop when USB is removed.  The '>r0!E' says: if recording stop now.

**Reply by spdavidb** [reply]
I'd like to record in 5.6k30 (360) with a MAX, saw in another thread that command is r5. Where does that get expressed in this code? Changing r0 to r5 seems to disable this script and the below is working, but capturing non-360 video now.

!MWAKE=2!MBOOT="!Lboot"!SAVEboot=<u0!X>u0<r0!S<u0>r0!E+!1O!R5

**Reply by dnewman-gpsw** [reply]
```<r0``` is not the same as ```r5```  one is a conditional on recording, the other is resolution change.

The camera will remember its last setting, so just set the resolution with the menu.  You don't need to set resolution all the time in the script.

```!MWAKE=2!MBOOT="!Lboot"!SAVEboot=<u0!X>u0<r0!S<u0!E+!1O!R5```

Does this above not work for you?

**Comment by AGuetz** [comment]
Hello. I am new to this platform and have to admit feeling out of place as I am not as knowledgeable when it comes to tech as I can tell you all are. I just purchased the HERO11 Mini with the hopes that I can get it to behave more like a dash cam after watching a YouTube video from someone claiming to have successfully done so and his link sent me to GitHub and why I am reaching out. Is there anyone who has been able to do this on the 11 Mini and willing to help me? Thanking you in advance :)

**Reply by dnewman-gpsw** [reply]
Not too hard.  Install the Labs firmware for your Mini from https://gopro.github.io/labs/  (this is a download and installation instructions.)

Then once Labs in installed, use the Mode button to turn on the camera and point it out this QR code
![image](https://github.com/gopro/labs/assets/10856084/e64ae02c-a5f6-45b3-b618-70b070da0d81)

This will save a boot script to your SD card.  If you every want to remove to script, delete the file MISC/boot or scan a code to erase the boot command.

It is extend version of the above example.

It has this logic:
- you start with shutter (Quick Capture) the camera will behave like all other Mini 11s.
- you power up (mode button) without USB power connected, it will behave all other Mini 11s.
- to change the camera (different), power up with the mode button, after the boot message plug-in USB power. The camera will now charge normally.
- Dashcam, on USB power is applied when the camera is off, it will boot, and start recording immediately, and it will stop recording and shutdown when power is removed.

**Comment by AGuetz** [comment]
Thank you David for your reply!

 

I did previously try to install that software (followed the instructions) and it basically corrupted my device to the point it would turn on, beep 3 times, turn off, and repeat until it would finally shut down. Had to reinstall original software to get my camera to work again. Either I installed the software incorrectly or there is something wrong with that script.

 

Aside from that issue, are you aware of any QR code that in addition to what you provided, actually gives the camera the command to turn on when vehicle motion is noted and turns off when vehicle is stationary? Also, any looping possibilities with the mini?

 

Thanks again!

**Reply by dnewman-gpsw** [reply]
There is no really corrupting a device, can't install wrong or have a bad firmware install.  What you can do is have messed up settings, or have script that is wrong.  Camera was likely doing what you told it to do.  With Labs you have significantly more power, and more ways to screw up.  The metadata in each file and the qrlog.txt log file will explain anything you might have done.  If is continue to misbehave, try a factory reset, and maybe share you qrlog.txt and metadata (use https://gopro.github.io/labs/control/metadata/ to extract metadata for you last video file.)  The above command works well on Mini, I just tested it.

**Comment by AGuetz** [comment]
Ok, will give it another go and follow your suggestions.

 

Thanks again!

**Comment by AGuetz** [comment]
Hello David,

 

Is that your YouTube video showing how you turned your 11 mini into a dash cam?

**Comment by dnewman-gpsw** [comment]
Likely I had one for MAX and one for Mini.

**Comment by AGuetz** [comment]
Install worked…thanks again!

 

Once I scan the QR code I should be good to go I presume? No additional changes/modifications I need to make in the camera settings?

**Reply by dnewman-gpsw** [reply]
You may need to set you video mode, and protune settings, as the script above uses the current settings.

**Comment by AGuetz** [comment]
I think I must have scanned the wrong QR code, and not yours, because it did boot up and start recording when connected to my USB port but in your video with your QR code, it would start recording only once your car was in motion, and automatically stopped recording when your vehicle was parked. I’m going to search for your YouTube video once again, and scan that QR code. Will I need to reinstall the software or should I be able to simply scan your QR code and override the one that I scanned?On Jan 3, 2024, at 5:51 PM, David Newman ***@***.***> wrote:﻿
You may need to set you video mode, and protune settings, as the script above uses the current settings.

**Reply by dnewman-gpsw** [reply]
These scripts are mini-programs, so you can make them do whatever you want (almost.) A more complex script might add a motion detection to the power trigger.  In general a scan will replace the previous command, as there can be only one boot command.  This particular script is really for hobbyist coders looking for more control and custom settings tailored to they're needs.  None of my examples are intended to be perfect solutions, rather starting points to create customization.

**Comment by AGuetz** [comment]
Thanks for the clarification. Unfortunately I would not know where to begin with creating what I am intending which is for my GoPro to behave like a dash cam. Your YouTube video is the closest I could find and you were kind enough to share your QR Code in that video so I’ll simply look for it and scan that code. Thanks again!!!Anthony GuetzoianBroker, DRE 012171066700 Fallbrook Ave., Suite 267|West Hills, ***@***.***     On Jan 3, 2024, at 6:11 PM, David Newman ***@***.***> wrote:﻿
These scripts are mini-programs, so you can make them do whatever you want (almost.) A more complex script might add a motion detection to the power trigger.  In general a scan will replace the previous command, as there can be only one boot command.  This particular script is really for hobbyist coders looking for more control and custom settings tailored to they're needs.  None of my examples are intended to be perfect solutions, rather starting points to create customization.
